app/models/intervention.rb

```diff
@@ -510,6 +511,13 @@ def update_temporality
     end
   end
 
+  # Set product name if its blank
+  def set_product_name(product,index)
```

Comment:
Instead of using 'index' which adds a gigantic amount of coupling, you should rely on the fact that "land_parcel" and "plant" share the same `InterventionGroup` (ex: "output.group.targets.first #=> land_parcel").

Reply:
@Aquaj ,
Actually I used .first concept, But Here add more options their user can select culture and land parcel multiple times, So .first is not working, That the reason I used Index here,

Reply:
In the case of a sowing intervention like here the structure is : 
```
Intervention --(many)-> InterventionGroup   --> InterventionTarget
                                             \-->  InterventionOutput
```
The association you use goes straight ignoring the groups to the outputs and targets, like :
```
Intervention --(many)-> InterventionOutputs
              \--(many)-> InterventionTargets
```
So when you are manipulating an `InterventionOutput` you can use that grouping hierarchy:
```
[ output ]
             Intervention
           /             \
       IGroup           IGroup
     /       \         /      \
> IOutput  ITarget    IOutput  ITarget
```
```
[ output.group ]
             Intervention
           /             \
    >  IGroup           IGroup
     /       \         /      \
 IOutput  ITarget    IOutput  ITarget
```
```
[ output.group.targets ]
             Intervention
           /             \
       IGroup           IGroup
     /       \         /      \
 IOutput  [ITarget]   IOutput  ITarget
```
```
[ output.group.targets.first ]
             Intervention
           /             \
      IGroup           IGroup
     /       \         /      \
 IOutput  ITarget <   IOutput  ITarget
```

What you refer to must be the `outputs` and `targets` on the `Intervention`, not on the `InterventionGroup` where you were using a structure looking like :
```
[ output ]
        Intervention
      ___ |      |_____
     /    |      |     \
    /     |      |      \
   /      |      |       \
>IOutput IOutput ITarget ITarget
```
```
[ output.intervention ]
      > Intervention
      ___ |      |_____
     /    |      |     \
    /     |      |      \
   /      |      |       \
IOutput IOutput ITarget ITarget
```
```
[ output.intervention.targets ]
        Intervention
      ___ |      |_____
     /    |      |     \
    /     |      |      \
   /      |      |       \
IOutput IOutput [ITarget ITarget]
```
```
[ output.intervention.targets.first ]
      > Intervention
      ___ |      |_____
     /    |      |     \
    /     |      |      \
   /      |      |       \
IOutput IOutput ITarget< ITarget 
```